packages/block-editor/src/components/block-list/index.js

```diff
@@ -46,16 +48,37 @@ function BlockList(
 			isDraggingBlocks,
 		} = select( 'core/block-editor' );
 
+		const selectedBlockClientId = getSelectedBlockClientId();
+		const multiSelectedBlockClientIds = getMultiSelectedBlockClientIds();
+
+		// Determine if there is a template part block to highlight.
+		const activeTemplatePartId = ( function () {
```

Comment:
It feels like this could be expensive to run all the time when template parts don't exist. I wonder if it would be possible to avoid running this code when the template part block has not been registered (That would keep it out of non-fse sites for now)

Reply:
> It feels like this could be expensive to run all the time when template parts don't exist.

I didn't think this would be very expensive?  And its up at the block-list level so its not running for every block.

> I wonder if it would be possible to avoid running this code when the template part block has not been registered

Yeah, I think we could gate it behind the experiment being enabled.

Reply:
> I didn't think this would be very expensive

I guess it'd just be expensive for deeply nested blocks, which would have to parse all the way up the block tree to get a result. But I think it's memoized so that helps. I also didn't realize it wasn't running for every block which is nice :) So it's probably me just being overly cautious. Definitely fine to try out in the site editor 👍

Reply:
Gated this to check for the site editor experiment - bea6a37